features/rt-cetsa-intensity-extraction-tool/src/polus/images/features/rt_cetsa_intensity_extraction/__init__.py

```diff
@@ -0,0 +1,286 @@
+
+    # build header
+    # check the first plate for the plate dimensions
+    nb_wells = len(measures[0][1])
+    plate_size = PlateSize(nb_wells)
+    plate_row_count = range(PLATE_DIMS[plate_size][0])
+    plate_col_count = range(PLATE_DIMS[plate_size][1])
+    plate_coords = itertools.product(plate_row_count, plate_col_count)
+
+    header = ["Temperature"] + [
+        alphanumeric_row(
+            row,
+            col,
+            (PLATE_DIMS[plate_size][0], PLATE_DIMS[plate_size][1]),
+        )
+        for row, col in plate_coords
+    ]
+
+    # build dataframe
+    # roundup temperature
+    rows = [[round(measure[0], 2), *measure[1]] for measure in measures]
+    df = pandas.DataFrame(rows, columns=header)
+
+    # Set the temperature as the index
+    df.set_index("Temperature", inplace=True)
+
+    # Sort the rows by temperature
+    df.sort_index(inplace=True)
+
+    return df
+
+
+def extract_wells_intensity(
+    image_path: pathlib.Path,
+    params_path: pathlib.Path,
+) -> list[int]:
+    """Extract well intensities from RT_CETSA image and mask.
+
+    Args:
+        image_path: Path to the RT_CETSA well plate image.
+        params_path: Path to the corresponding params file.
+
+    Returns:
+        corrected intensity for each well.
+    """
+    with bfio.BioReader(image_path) as reader:
+        image = reader[:]
+    with params_path.open("r") as f:
+        params = PlateParams(**from_json(f.read()))
+
+    intensities = []
+
+    for y, x in itertools.product(range(len(params.Y)), range(len(params.X))):
+        intensity = extract_intensity(
+            image,
+            params.X[x],
+            params.Y[y],
+            params.roi_radius,
+        )
+        intensities.append(intensity)
+
+    return intensities
+
+
+def extract_intensity(image: np.ndarray, x: int, y: int, r: int) -> int:
+    """Get the well intensity
+
+    Args:
+        image: _description_
+        x: x-position of the well centerpoint
+        y: y-position of the well centerpoint
+        r: radius of the circle inscribed in the square area of interest.
+
+    Returns:
+        int: The background corrected mean well intensity
+    """
+    # we take a square area around the well center
+    x_min = max(x - r, 0)
+    x_max = min(x + r, image.shape[1])
+    y_min = max(y - r, 0)
+    y_max = min(y + r, image.shape[0])
+
+    patch = image[y_min:y_max, x_min:x_max]
+    background = patch.ravel()
+    background.sort()
+
+    # Subtract lowest pixel values from average patch pixel values
+    return int(np.mean(patch) - np.median(background[: int(0.05 * background.size)]))
+
+
+def extract_wells_intensity_from_mask(
+    image_path: pathlib.Path,
+    mask_path: pathlib.Path,
+) -> list[int]:
+    """Extract well intensities from RT_CETSA images using a labeled mask.
+
+    This method is degree of magnitude slower than extract_wells_intensity
+    and is just provided for convenience. Consider using extract_wells_intensity instead.
```

Comment:
I am surprised that numpy indexing with a mask is so much slower. I'll have to keep that in mind if we can assume that masks always have a specific shape